tools/cgeist/Test/Verification/base_cast.cpp

```diff
@@ -0,0 +1,76 @@
+// RUN: cgeist %s --function=* -S | FileCheck %s
+// RUN: cgeist %s --function=* --struct-abi=0 -memref-abi=0 -S | FileCheck %s --check-prefix CHECK-STR
+
+
+struct A {
+  int val1;
+  int val2;
+};
+
+struct B {
+  bool bool1;
+};
+
+struct C : B, A {
+  int val3;
+};
+
+struct D : C {
+  int val3;
+  bool bool2;
+};
+
+C* castAtoC(A *a) {
+  // A -> C
+  return static_cast<C *>(a);
+};
+
+D* castBtoD(B *b) {
+  // B -> C -> D
+  return static_cast<D *>(b);
+}
+
+D* castAtoD(A *b) {
+  // A -> C -> D
```

Comment:
Added two inheritance chains one of the (A->C->D) requires accessing the base A at an offset.